components/style/values/specified/length.rs

```diff
@@ -458,6 +458,17 @@ impl Parse for Length {
     }
 }
 
+impl Either<Length, None_> {
+    #[inline]
+    #[allow(missing_docs)]
+    pub fn parse_non_negative_length(context: &ParserContext, input: &mut Parser) -> Result<Self, ()> {
+        if input.try(|input| None_::parse(context, input)).is_ok() {
+            return Ok(Either::Second(None_));
+        }
+        Length::parse_internal(input, AllowedNumericType::NonNegative).map(Either::First)
```

Comment:
I don't think it's worth. We may have other stuff (see `LengthOrNumber`) where we parse non-negative stuff but _any_ of those may be non-negative. But maybe it's ok.